### PR TITLE
feat: cfd-568 - migrate new naptan site

### PR DESCRIPTION
**Pull request by JwanKhalaf**
# Description

-   Handle the new URL and refactor the naptan handler.

# Testing instructions

-   Run the handler via the AWS Console

# Type of change

-   [x] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

### CODE REVIEW

File: repos/fdbt-reference-data-service/src/retrievers/csv_retriever/main.py [hunk 1]
```diff
@@ -2,31 +2,70 @@
...
+        data_url = os.getenv('DATA_URL')
+        backup_data_url = os.getenv('BACKUP_DATA_URL')
+        content_type = os.getenv('CONTENT_TYPE')
+        bucket_name = os.getenv('BUCKET_NAME')
+
+        if data_url is None:
+            raise Exception("There was no DATA_URL environment variable!")
+
+        if backup_data_url is None:
+            raise Exception("There was no BACKUP_DATA_URL environment variable!")
```
Comment (jacsamell) [inline repos/fdbt-reference-data-service/src/retrievers/csv_retriever/main.py:24]: This will break the NocRetriever

**Review by jacsamell — CHANGES_REQUESTED** [review]

File: repos/fdbt-reference-data-service/src/retrievers/serverless.yml [hunk 2]
```diff
@@ -1,199 +1,200 @@
...
-              treatMissingData: notBreaching
-            - name: functionInvocations
-              treatMissingData: notBreaching
+  NaptanRetriever:
+    handler: csv_retriever/main.lambda_handler
+    timeout: 60
+    memorySize: 512
+    environment:
+      DATA_URL: https://naptan.dft.gov.uk/Download/File/Stops.csv
+      BACKUP_DATA_URL: https://beta-naptan.dft.gov.uk/Download/File/Stops.csv
```
Comment (jacsamell) [inline repos/fdbt-reference-data-service/src/retrievers/serverless.yml:119]: These should be the other way around I think, the beta one is the one that currently works

File: repos/fdbt-reference-data-service/src/retrievers/.python-version [hunk 3]
```diff
@@ -0,0 +1 @@
+jtest
```
Comment (jacsamell) [inline repos/fdbt-reference-data-service/src/retrievers/.python-version:1]: whats this?

File: repos/fdbt-reference-data-service/src/retrievers/csv_retriever/main.py [hunk 4]
```diff
@@ -2,31 +2,70 @@
...
+                            'ContentType': content_type 
+                        }
+                    )
+        else:
+            logger.info("File retrieved is not a ZIP file")
+
+            target_file = os.getenv('TARGET_FILE')
+
+            if target_file is None:
+                raise Exception("There was no TARGET_FILE environment variable!")
```
Comment (matt-twomey) [inline repos/fdbt-reference-data-service/src/retrievers/csv_retriever/main.py:65]: It looks like target_file may not be set sometimes, as in line 50 above, and we don't raise an exception. Are we sure we want to except if there's no target_file environment variable, in the case that the file isn't a zip file?

File: repos/fdbt-reference-data-service/src/retrievers/serverless.yml [hunk 5]
```diff
@@ -1,199 +1,200 @@
...
+  name: aws
+  region: eu-west-2
+  runtime: python3.8
+  stage: ${opt:stage, 'test'}
+  stackName: ${self:service}-${self:provider.stage}
+  lambdaHashingVersion: 20201221
+  iam:
+    role:
+      statements:
+        - Effect: "Allow"
```
Comment (matt-twomey) [inline repos/fdbt-reference-data-service/src/retrievers/serverless.yml:13]: Are the formatting changes intentional, or the result of an IDE not using the correct formatter?

File: repos/fdbt-reference-data-service/src/retrievers/csv_retriever/main.py (hunk 1 above)
Comment (Aaron-Collis-IW) [reply repos/fdbt-reference-data-service/src/retrievers/csv_retriever/main.py:24]: MOved this to with in the except

File: repos/fdbt-reference-data-service/src/retrievers/csv_retriever/main.py (hunk 4 above)
Comment (JwanKhalaf) [reply repos/fdbt-reference-data-service/src/retrievers/csv_retriever/main.py:65]: On line 53, it checks for it.

File: repos/fdbt-reference-data-service/src/retrievers/serverless.yml (hunk 2 above)
Comment (Aaron-Collis-IW) [reply repos/fdbt-reference-data-service/src/retrievers/serverless.yml:119]: The beta one is the one that works but if we swap them around we will use the beta link until it dies, wouldn't it be better to switch to the normal one as soon as it goes live, which is what the current code will do. Does that make sense?

**Review by Aaron-Collis-IW — APPROVED** [review]
Looks good, just need to wait for locality name to be added by dft

**Review by joeds13 — APPROVED** [review]

**Pull request merged**